dask/array/core.py

```diff
@@ -1270,6 +1270,29 @@ def coarsen(reduction, x, axes):
     return Array(merge(x.dask, dsk), name, blockdims=blockdims, dtype=dt)
 
 
+@wraps(chunk.broadcast_to)
+def broadcast_to(x, shape):
+    shape = tuple(shape)
+    ndim_new = len(shape) - x.ndim
+    if ndim_new < 0 or any(new != old
+                           for new, old in zip(shape[ndim_new:], x.shape)
+                           if old != 1):
+        raise ValueError('cannot broadcast shape %s to shape %s'
+                         % (x.shape, shape))
+
+    name = next(names)
+    blockdims = (tuple((s,) for s in shape[:ndim_new])
```

Comment:
I was thinking about this to. Arguably, this function should take optional `blockshape` and `blockdims` arguments.

I do think (auto-)reblocking will cover most of that use case. In particular, there is never a concern about memory usage with the new, arbitrarily large chunks will always be broadcast versions of the  original chunks.

One potential downside of separate reblocking is that it wouldn't be able to take advantage of the fact that many of the chunks would be redundant. So that would entail somewhat more overhead, both for the scheduler and for slicing the broadcast expanded dimensions with numpy.

I'll make a follow-up issue.